packages/shorebird_cli/lib/src/shorebird_android_artifacts.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:path/path.dart' as p;
+import 'package:scoped/scoped.dart';
+import 'package:shorebird_cli/src/command.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+
+/// Throw when multiple artifacts are found in the build directory.
+class MultipleArtifactsFoundException implements Exception {
+  MultipleArtifactsFoundException({
+    required this.buildDir,
+    required this.foundArtifacts,
+  });
+
+  final String buildDir;
+  final List<FileSystemEntity> foundArtifacts;
+
+  @override
+  String toString() {
+    return 'Multiple artifacts found in $buildDir: '
+        '${foundArtifacts.map((e) => e.path)}';
+  }
+}
+
+extension on String {
+  String get artifactId => replaceAll(RegExp(r'\W'), '').toLowerCase();
+}
```

Comment:
I believe this makes it anonymous and thus local to this file, correct?

Reply:
Exactly!

Reply:
It would be nice to add a comment here explaining what this is doing and why we're using this method to compute an artifact id.

Reply:
Did you push the commit that adds the comment? I'm not seeing it on my end. Something like "Removes all non-word characters (i.e., `[^a-zA-Z0-9_]`) from the string" would be great.

Reply:
Also mentioning the "why" is great too – I feel like we've had more cases of "why did we do that again?" this week than usual.

Reply:
@bryanoltman I pushed yes! I can see the comment in my PR, can you check again please?

Reply:
Yep, see it now. Added some suggestions 👍